src/components/lib/cos_kernel_api.c

```diff
@@ -421,7 +422,7 @@ cos_captbl_alloc(struct cos_compinfo *ci)
 	assert(ci);
 
 	if (__alloc_mem_cap(ci, CAP_CAPTBL, &kmem, &cap)) return 0;
-	if (call_cap_op(ci->captbl_cap, CAPTBL_OP_CAPTBLACTIVATE, cap, ci->pgtbl_cap, kmem, 0)) BUG();
+	if (call_cap_op(ci->captbl_cap, CAPTBL_OP_CAPTBLACTIVATE, cap, __compinfo_metacap(ci)->local_mi.pgtbl_cap, kmem, 0)) BUG();
```

Comment:
I mean, all the `ci->pgtbl_cap` which is `SELF_PT` seem to get replaced with `local_mi.pgtbl_cap` (`UNTYPED_PT`), so I said that. 
But your last comment about `SELF_PT` having only virtual memory of the booter, I'm trying to understand that. In that case booter and/or memmgr, should have this distinction of `SELF_PT` and `UNTYPED_PT` and all other components only refer to `SELF_PT`? I'll look at the source code again to see how all this fits.